### PR TITLE
simple test for no grab = no bites

### PR DESCRIPTION
**Pull request by DoctorGoat**
It's not complex. It's not tested. It has no style. It has no grace. This PR, needs.. testing and ideas. 

More notes:
This has a lot of room for expansion. Scouring #mods-and-development on the discord for a while will help. 

The second step of this is going to be making exhaustion/pain/more grabs raise bite chance, so more zombies are still a serious problem.

The goal is to make individual zombies less of a threat unless you let them grab you and don't try to break the grab.

### CODE REVIEW

**Comment by Firestorm01X2** [comment]
You've created PR from you master branch. It may give extra problems if you will try to manage multiple PR at the same time.

Next time create branch and create PR from it. It s much more managable.

**Comment by DoctorGoat** [comment]
I have, unfortunately, noticed. I will probably have to work something out, since I have a few different things going on.

**Comment by DoctorGoat** [comment]
I've been testing this, and with a set of lucky chain mail at the start, my survivor was able to go four days without a bite by staying the hell away from crowds. I don't know how it would have gone without the chainmail, but it's soaking up enough damage that I don't feel endangered by single zombies, at least.

Which makes sense for a fairly high end set?

File: src/mattack_actors.cpp [hunk 1]
```diff
@@ -294,7 +295,11 @@ void bite_actor::load_internal( JsonObject &obj, const std::string &src )
 void bite_actor::on_damage( monster &z, Creature &target, dealt_damage_instance &dealt ) const
 {
     melee_actor::on_damage( z, target, dealt );
-    if( one_in( no_infection_chance - dealt.total_damage() ) ) {
+    bool infectionchance = one_in( no_infection_chance - dealt.total_damage() );
+    //disable biting on ungrabbed targets
+    if( !target.has_effect( effect_grabbed ) ) {
+        infectionchance = 0;
```
Comment (BevapDin) [inline src/mattack_actors.cpp]: `infectionchance` is declared as a `bool`, don't assign an integer to it, assign an boolean value (`true` or `false`) to it.

Also: the `else if` branch makes setting `infectionchance` here useless, it is never read after that assignment, so the assignment does not matter.

Also: the name is odd, it's not an "infection chance" (that would be some numeric value, a "X in Y" value), but this is more a "should infection actual happen" value. I suggest a simpler version: `if( has_effect( ... ) && one_in( ... ) ) {` where you don't even need the variable.

Comment (DoctorGoat) [reply src/mattack_actors.cpp]: there was some weirdness i was stumbling into early on in doing this little PoC where I didn't notice that one_in was returning a bool and I was also juggling trying to figure out why there are three enemies that have specific infection chance fields in the json when the rest don't, and I believe the former is the cause for the weirdness in typing.

This is definitely an improvement on what I did, though.

**Pull request merged**